cs/src/core/Index/Common/Contexts.cs

```diff
@@ -428,15 +428,16 @@ public void Initialize(StreamReader reader)
         /// <param name="token"></param>
         /// <param name="checkpointManager"></param>
         /// <param name="deltaLog"></param>
-        /// <returns></returns>
-        internal void Recover(Guid token, ICheckpointManager checkpointManager, DeltaLog deltaLog = null)
+        /// <returns> Any user-specified commit cookie written as part of the checkpoint </returns>
+        internal byte[] Recover(Guid token, ICheckpointManager checkpointManager, DeltaLog deltaLog = null)
```

Comment:
Better to be an overload of recover that returns void, but includes an `out commitCookie` parameter so it is obvious what the array is.